samples/shadow/shadow_sync/main.cpp

```diff
@@ -80,20 +80,19 @@ static void s_changeShadowValue(
 {
     fprintf(stdout, "Changing local shadow value to %s.\n", value.c_str());
 
+    ShadowState state;
+    JsonObject desired;
+    desired.WithString(shadowProperty, value);
+    JsonObject reported;
+    reported.WithString(shadowProperty, value);
+    state.Desired = desired;
+    state.Reported = reported;
+    
     UpdateShadowRequest updateShadowRequest;
     Aws::Crt::UUID uuid;
     updateShadowRequest.ClientToken = uuid.ToString();
-
-    JsonObject stateDocument;
-    JsonObject reported;
-    reported.WithString(shadowProperty, value);
-    stateDocument.WithObject("reported", std::move(reported));
-    JsonObject desired;
-    desired.WithString(shadowProperty, value);
-    stateDocument.WithObject("desired", std::move(desired));
-
-    updateShadowRequest.State = std::move(stateDocument);
-    updateShadowRequest.ThingName = thingName;
+    updateShadowRequest.ThingName = thingName.c_str();
```

Comment:
why use c_str() here?

Reply:
@JonathanHenson 

Yes here it is not required.

We are using it our program because thingName is **std:string** and updateShadowRequest.ThingName is **Aws::Crt::String**. if dont use c_str() it is failing to compile.

Anyway in the sample program it is not required, fixed it.